prime-angular-frontend/src/app/modules/auth/shared/services/permission.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+
+import { Role } from '@auth/shared/enum/role.enum';
+
+import { AccessTokenService } from '@auth/shared/services/access-token.service';
+
+export interface IPermissionService {
+  hasRoles(...roles: Role[]): boolean;
+  hasAnyRole(...roles: Role[]): boolean;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PermissionService implements IPermissionService {
+
+  constructor(
+    private accessTokenService: AccessTokenService
+  ) { }
+
+  public hasRoles(...roles: Role[]): boolean {
```

Comment:
When you use the rest operator (`...roles`) you're saying that if I pass in `hasRoles(Role.1, Role.2, Role.3, ..., Role.N)` that you want it as an array of roles, but if I pass in an array of Roles `hasRoles([Role.1, Role.2, Role.3, ..., Role.N])` you would get `[[Role.1, Role.2, Role.3, ..., Role.N]]` so a developer consuming your method would need to know they have to invoke it like `hasRoles(...[Role.1, Role.2, Role.3, ..., Role.N]);`.  The rest operator is great in methods if you need a variadic method with infinity arity, but in this case it would be easier for developers to consume if you manage the difference between singular and multiples of a type within the method:

```js
public hasRoles(roles: Role | Role[]): boolean {
  roles = (Array.isArray(roles)) ? roles : [roles];
  const assignedRoles = this.accessTokenService.roles();
  return roles.every(r => assignedRoles.includes(r));
}
```